plumbing/format/packfile/delta_selector.go

```diff
@@ -261,6 +267,16 @@ func (dw *deltaSelector) walk(
 }
 
 func (dw *deltaSelector) tryToDeltify(indexMap map[plumbing.Hash]*deltaIndex, base, target *ObjectToPack) error {
+	// Original object might not be present if we're reusing a delta, so we
+	// ensure it is restored.
+	if err := dw.restoreOriginal(target); err != nil {
```

Comment:
isn't this making it actually slower than before (even if it uses way less memory)?

Reply:
I've just run `borges pack` with repository https://github.com/numpy/numpy (downloaded locally) and the results are quite close.

* Original version:
  * 203.83user 12.34system 2:28.80elapsed 2.4 Gb RAM
  * 204.40user 12.27system 2:29.30elapsed 2.4 Gb RAM
* With proposed changes:
  * 217.82user 11.97system 2:25.72elapsed 1 Gb RAM
  * 220.06user 11.81system 2:26.28elapsed 1 Gb RAM

It seems that it's a bit slower (user time is bigger) but it spends more system time. Maybe heap allocation.